### PR TITLE
Correcting library path of hsakmt-roct for Centos7 and cleaning up the patch file.

### PR DESCRIPTION
**Pull request by renjithravindrankannath**
RVS build failing on Centos7 since library path for hsakmt-roct is hsakmt-roct-x.y.z-abcdefg/lib64/
Added check for the directory name to set the correct library path.

### CODE REVIEW

**Comment by renjithravindrankannath** [comment]
Successfully built on Centos with the changes and all test passed.
`==> Installing rocm-validation-suite-5.0.2-4zyxn4r73qi6qz4tw6vjs3i7u4b4477w
==> No binary for rocm-validation-suite-5.0.2-4zyxn4r73qi6qz4tw6vjs3i7u4b4477w found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/f2/f249fe700a5a96c6dabf12130a3e366ae6025fe1442a5d11d08801d6c0265af4.tar.gz
==> Applied patch /spack/var/spack/repos/builtin/packages/rocm-validation-suite/005-cleanup-path-reference-donot-download-googletest-yaml.patch
==> Applied patch /spack/var/spack/repos/builtin/packages/rocm-validation-suite/006-library-path.patch
==> Ran patch() for rocm-validation-suite
==> rocm-validation-suite: Executing phase: 'cmake'
==> rocm-validation-suite: Executing phase: 'build'
==> rocm-validation-suite: Executing phase: 'install'
==> rocm-validation-suite: Successfully installed rocm-validation-suite-5.0.2-4zyxn4r73qi6qz4tw6vjs3i7u4b4477w
  Fetch: 0.01s.  Build: 40.11s.  Total: 40.12s.
[+] /spack/opt/spack/linux-centos7-skylake_avx512/gcc-7.3.1/rocm-validation-suite-5.0.2-4zyxn4r73qi6qz4tw6vjs3i7u4b4477w
bash-4.2#`


`bash-4.2# cd /spack/opt/spack/linux-centos7-skylake_avx512/gcc-7.3.1/rocm-validation-suite-5.0.2-4zyxn4r73qi6qz4tw6vjs3i7u4b4477w/rvs/
bash-4.2# ls
conf                   libgm.so.0            libgst.so            libiet.so.0.0.50100  libpebb.so.0          libperf.so            libpesm.so.0.0.50100  librcqt.so.0          rvs
libbabel.so            libgm.so.0.0.50100    libgst.so.0          libmem.so            libpebb.so.0.0.50100  libperf.so.0          libpqt.so             librcqt.so.0.0.50100  testscripts
libbabel.so.0          libgpup.so            libgst.so.0.0.50100  libmem.so.0          libpeqt.so            libperf.so.0.0.50100  libpqt.so.0           libsmqt.so            userguide
libbabel.so.0.0.50100  libgpup.so.0          libiet.so            libmem.so.0.0.50100  libpeqt.so.0          libpesm.so            libpqt.so.0.0.50100   libsmqt.so.0
libgm.so               libgpup.so.0.0.50100  libiet.so.0          libpebb.so           libpeqt.so.0.0.50100  libpesm.so.0          librcqt.so            libsmqt.so.0.0.50100
bash-4.2#
`

**Comment by renjithravindrankannath** [comment]
@tldahlgren & @srekolam  Verified the changes on Centos7. Please review and merge the change.

File: var/spack/repos/builtin/packages/rocm-validation-suite/006-library-path.patch
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-validation-suite/006-library-path.patch:73]: FWIW It looks like there is a tab here on line 73 instead of spaces.

File: var/spack/repos/builtin/packages/rocm-validation-suite/006-library-path.patch
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-validation-suite/006-library-path.patch:115]: FWIW Same tab versus spaces here at line 115 PLUS 139, 146, 184, 191, 202, 227, 234, 258, 265, 276, 321, and 328 as well.

File: var/spack/repos/builtin/packages/rocm-validation-suite/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-validation-suite/package.py:46]: Since this appears to be an updated version of `007-library-path.patch` and the `007` version isn't used, should that file be removed as part of this PR?

**Review by tldahlgren — COMMENTED** [review]
Mainly a question about the patch.

File: var/spack/repos/builtin/packages/rocm-validation-suite/package.py
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocm-validation-suite/package.py:46]: Thank you for reviewing Tamara. 007-library-path.patch will be no longer needed.
I am removing 007-library-path.patch.

**Review by tldahlgren — APPROVED** [review]
LGTM. Deferring merge decision until one of the maintainers has a chance to comment.

**Comment by tldahlgren** [comment]
@arjun-raj-kuppala @srekolam Did you want to comment on this PR before it is merged?

**Review by srekolam — APPROVED** [review]

**Pull request merged**
